spyro/receivers/Receivers.py

```diff
@@ -343,13 +340,13 @@ def __func_build_cell_tabulations_3D(self):
 
         return cell_tabulations
 
-    def apply_source_receivers(self, rhs_forcing, residual, IT, is_local):
+    def apply_source_receivers(self, rhs_forcing, residual, IT):
         """
         The adjoint operation of interpolation (injection)
         """
         for rid in range(self.num_receivers):
             value = residual[IT][rid]
-            if is_local[rid]:
+            if self.is_local[rid]:
```

Comment:
Note that you'll have to call the `rhs_forcing.dat.data_with_halos` the same number of times (i.e., num_receivers times) in order to avoid deadlocks in parallel.